experimental/parser/testdata/parser/lists.proto.stderr.txt

```diff
@@ -484,30 +484,12 @@ error: missing message field tag in declaration
 54 |     map<int, int> x;
    |     ^^^^^^^^^^^^^^^^
 
-error: unexpected non-comparable type in map key type
-  --> testdata/parser/lists.proto:54:9
-   |
-54 |     map<int, int> x;
-   |         ^^^
-   = help: a map key must be one of the following types: int32, int64, uint32,
-           uint64, sint32, sint64, fixed32, fixed64, sfixed32, sfixed64, bool,
-           string
-
```

Comment:
I just made a comment on a diagnostic related to when the key type is not an allowed type, so I'm a little confused to not see it here. I would expect to see something here either about `int` not being a known type or about it not being a comparable type (which I just suggested we tweak the language). So I am quite surprised to neither here. What's that about?

Reply:
I think that legalization logic got deferred to the lowering step, and so the parser wouldn't be responsible for it anymore?